subscription_manager/controller.py

```diff
@@ -64,34 +69,35 @@ def delete_subscription(self, subscription_id: int) -> int:
         """
         pass
 
-    def get_subscription_by_id(self, subscription_id: int) -> dict:
+    def get_subscription_by_name(self, subscription_name: str) -> Subscription:
         """
-        Return subscription by identifier
+        Return subscription by name
         Returns:
-            dict: found subscription in dict representation
+            <class Subscription>: found subscription like Subscription object
         Raises:
             SubsNotFoundException: when subscription with this id doesn't exist in database
         """
-        pass
+        if type(subscription_name) is not str:
+            raise WrongTypeException(
+                f"Found wrong field 'subscription_name' type, expected:<str>, received:{type(subscription_name), subscription_name}"
+            )
+        if len(subscription_name) < 1:
+            raise InvalidValueException(
+                "Subscription name length should be more than one"
+            )
+        recieved_subscription: dict = self.dbhelper.get_subscription(subscription_name)
+        recieved_subscription.pop("_id")
```

Comment:
Is that a reason you need id, instead of having name as id ?

Reply:
Subscription object does not contain _id field, but database returns it, so I remove this field from database answer for returning a correct dict for Subscription class.